sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -310,8 +315,84 @@ public <T> Mono<CosmosItemResponse<T>> createItem(T item, CosmosItemRequestOptio
     }
 
+    private <T> Mono<CosmosItemResponse<T>> createItemWithRetriesInternal(
+        T item, CosmosItemRequestOptions options) {
+
+        return createItemInternal(item, options)
+            .onErrorResume(throwable -> {
+                Throwable error = throwable instanceof CompletionException ? throwable.getCause() : throwable;
+
+                if (!(error instanceof CosmosException)) {
+
+                    Exception nonCosmosException =
+                        error instanceof Exception ? (Exception) error : new RuntimeException(error);
+                    return Mono.error(nonCosmosException);
+                }
+
+                assert error instanceof CosmosException;
+                CosmosException cosmosException = (CosmosException) error;
+
+                if (cosmosException.getStatusCode() != HttpConstants.StatusCodes.CONFLICT) {
+                    return Mono.error(cosmosException);
+                }
+
+                InternalObjectNode internalObjectNode = InternalObjectNode.fromObjectToInternalObjectNode(item);
+                String itemId = internalObjectNode.getId();
+                RequestOptions requestOptions = ModelBridgeInternal.toRequestOptions(options);
+                @SuppressWarnings("unchecked")
+                Class<T> itemType = (Class<T>) item.getClass();
+
+                Mono<CosmosItemResponse<T>> readMono =
+                    this.getDatabase().getDocClientWrapper()
+                        .readDocument(getItemLink(itemId), requestOptions)
+                        .map(response -> {
+                            CosmosDiagnostics responseDiagnostics = response.getDiagnostics();
+                            if (responseDiagnostics != null &&
+                                responseDiagnostics.getClientSideRequestStatisticsRaw() != null) {
+
+                                CosmosDiagnostics errorDiagnostics = cosmosException.getDiagnostics();
+                                if (errorDiagnostics != null) {
+                                    responseDiagnostics.clientSideRequestStatistics().recordContributingPointOperation(
+                                        errorDiagnostics.getClientSideRequestStatisticsRaw()
+                                    );
+                                }
+
+                                response.addRequestCharge(cosmosException.getRequestCharge());
+                            }
+                            return ModelBridgeInternal
+                                .createCosmosAsyncItemResponse(response, itemType, getItemDeserializer());
+                        })
+                        .single();
+
+                return readMono
+                    .onErrorMap(readThrowable -> cosmosException)
+                    .flatMap(readResponse -> {
+                        if (readResponse.getStatusCode() == 200) {
+                            return Mono.just(itemResponseAccessor.withRemappedStatusCode(
```

Comment:
Are we still going to remove the response body based on `contentResponseOnWriteEnabled`?

Reply:
Missed that - good catch